### ISSUE TITLE
PACKAGE_PARSER do not work with UTF-8

### ISSUE THREAD

**Issue by virosh** [fluffos/fluffos#660]
Today i found that the package PARSER does not work with UTF-8 encoding.
I found this during testing support for Cyrillic characters in verbs for communication - say, shout, etc.
What i found it that if the characters are not in English (probably latin-1), they are discarded.
After some debugging i decided to create a dummy command for 'say' command and test and everything is working there.
So for some reason the package PARSER is not handling encodings at all.

Debug output from the commands follow:
From verbs:
```
(HP:24 SP:26)/cmds/user> parse -v say blah
Parsing say blah...
Trying interpretation: say:blah:
Rule: STR
Trying rule: STR
  parse_rule
  Adding match: tok = 2 start = 1 end = 1
  State is: aec762f0, num_errors: 0

  Taking rest of sentence as STR
    parse_rule
      we_are_finished
      Trying can_say_str ... (/std/user#30)
      Trying can_say_str ... (/std/user#30)
      Trying can_verb_str ... (/std/user#30)
      Trying can_verb_rule ... (/std/user#30)
      Trying can_say_str ... (/verbs/common/say)
      Return value was: 1
      Saving successful match: do_say_str (verbs/common/say)
    exiting parse_rule ...
  Done trying to match STR
Rule:
Calling do_say_str ...
You say: blah
```

```
(HP:24 SP:26)/cmds/user> parse -v say блах
Parsing say блах...
Trying interpretation: say:
Rule: STR
Trying rule: STR
  parse_rule
  Ran out of words to parse.
Rule:
Trying rule:
  parse_rule
    we_are_finished
    Trying can_say ... (/std/user#30)
    Trying can_say ... (/std/user#30)
    Trying can_verb ... (/std/user#30)
    Trying can_verb_rule ... (/std/user#30)
    Trying can_say ... (/verbs/common/say)
    Return value was: 1
    Saving successful match: do_say (verbs/common/say)
  exiting parse_rule ...
Calling do_say ...
You mutter to yourself.
```

From command:
```
(HP:24 SP:26)/cmds/user> csay blah
You say: blah
```
```
(HP:24 SP:26)/cmds/user> csay блах
You say: блах
```

**Comment by thefallentree** [comment]
why did you close it?

**Comment by virosh** [comment]
After some reading of the code i realized the PARSER package was created and written with the intentions to work with only English language and it was never completed fully. I know that this have some limitations when you wish to use "verbs" in other languages, but i also don't think it will be possible to make this package to support multi-language as you need to add the parsing capabilities inside the package for every language you want to support...
I also realized that in my situation it's easier and better to just remove all verbal commands from the "verbs" system and create them as normal commands.

**Comment by thefallentree** [comment]
yeah, the grammar part is not going to work, but I'm surprised "say XXX"
doesn't work either

>

**Comment by virosh** [comment]
I haven't have time to work on my mudlib for some time and after some reconsideration i think you are right and this should work so i am reopening the issue.

**Comment by thefallentree** [comment]
What exactly is the issue you are reporting?

**Comment by virosh** [comment]
The issue is that if you are using PACKAGE_PARSER and verbs as a commands like "say" it works only if you are saying something using English language.
You can see my first post for example what i mean. The example is with cyrillic.
In my case i could easily rewrite all conversation verbs to commands and fix this, however i think that this should work.

This is the reason i opened the issue in the first place, then decided to rewrite all verbs to commands and closed it.
However i haven't have any time to work on this project and now when i have some free time and reviewed this again i am reconsidering that the verbs should be able to work not only with English. 
And again i want to make it clear that i am not speaking for the grammar part of the parser.

Also i am not sure if this can be fixed at all. Unfortunately i do not understand C or C++ and the code of the driver and how it works is very much cryptic to me. This is the reason to not be able to try fix this by myself and opening the issue.
